lib/api.js

```diff
@@ -26,8 +26,8 @@ function buildRequest(apiHost, apiPort, method, options, requestOptions, callbac
 
     sendOptions.method = (method === 'del' ? 'delete' : method).toUpperCase();
     sendOptions.uri = url.format({
-        protocol: 'http',
-        hostname: apiHost,
+        protocol: url.parse(apiHost).protocol || 'http',
+        hostname: url.parse(apiHost).host || apiHost,
```

Comment:
I'll be honest that it took me a moment to figure out what was going on here. The key part of it is, is that `url.parse().host` will return falsely if `apiHost` is not a valid full URL.

It might be nice to add a test to explicitly demonstrate this behaviour, but it's SNABD